dash-pipeline/SAI/templates/saifixedapis.cpp.j2

```diff
@@ -2,136 +2,130 @@
-#include <stdio.h>
-#include <string.h>
-#include <PI/pi.h>
-#include <grpcpp/grpcpp.h>
-#include "p4/v1/p4runtime.grpc.pb.h"
 #include "utils.h"
-extern "C" {
-#include "sai.h"
-#include "saiobject.h"
-#include "saistatus.h"
-#include "saitypes.h"
-#include "saiextensions.h"
-#include "sairoute.h"            // TODO: Change to template generated values
-}
-#include <fstream>
-#include <google/protobuf/text_format.h>
-#include <google/rpc/code.pb.h>
-
-#if __APPLE__
-#include <net/ethernet.h>
-#else
-#include <netinet/ether.h>
-#include <netinet/in.h>
-#endif
-
-using namespace std;
+#include "saiimpl.h"
 
 // TODO to be moved to separate class/namespace
 
 static std::shared_ptr<grpc::Channel> _grpcChannel;
 extern std::unique_ptr<p4::v1::P4Runtime::Stub> stub;
 
-#define DASH_BMV2_NUM_PORTS 2 
+#define DASH_BMV2_NUM_PORTS 2
 #define DASH_BMV2_CPU_QOS_NUMBER_OF_QUEUES 0
 
-#define OJBTYPE_SHFT 48
-#define MAKE_OID(_objtype, _objval) (sai_object_id_t)(((sai_object_id_t)_objtype<<OJBTYPE_SHFT)+(sai_object_id_t)_objval)
+#define DASH_OBJECT_SHFT 48
+#define DASH_MAKE_OID(_objtype, _objval) (sai_object_id_t)(((sai_object_id_t)_objtype<<DASH_OBJECT_SHFT)+(sai_object_id_t)_objval)
 
-#define DASH_BMV2_DEFAULT_CPU_PORT_ID MAKE_OID(SAI_OBJECT_TYPE_PORT,64)
-#define DASH_BMV2_DEFAULT_VLAN_ID MAKE_OID(SAI_OBJECT_TYPE_VLAN,1)
-#define DASH_BMV2_DEFAULT_VRF_ID MAKE_OID(SAI_OBJECT_TYPE_VIRTUAL_ROUTER,1)
-#define DASH_BMV2_DEFAULT_1Q_BRIDGE_ID MAKE_OID(SAI_OBJECT_TYPE_BRIDGE,1)
+#define DASH_BMV2_DEFAULT_CPU_PORT_ID DASH_MAKE_OID(SAI_OBJECT_TYPE_PORT,64)
+#define DASH_BMV2_DEFAULT_VLAN_ID DASH_MAKE_OID(SAI_OBJECT_TYPE_VLAN,1)
+#define DASH_BMV2_DEFAULT_VRF_ID DASH_MAKE_OID(SAI_OBJECT_TYPE_VIRTUAL_ROUTER,1)
+#define DASH_BMV2_DEFAULT_1Q_BRIDGE_ID DASH_MAKE_OID(SAI_OBJECT_TYPE_BRIDGE,1)
 
-sai_status_t sai_create_switch_dummy(
+static sai_object_id_t port_list[DASH_BMV2_NUM_PORTS] = {
+    DASH_MAKE_OID(SAI_OBJECT_TYPE_PORT,1),
+    DASH_MAKE_OID(SAI_OBJECT_TYPE_PORT,2)
+};
+
+static sai_status_t dash_sai_create_switch(
         _Out_ sai_object_id_t *switch_id,
         _In_ uint32_t attr_count,
-        _In_ const sai_attribute_t *attr_list) {
+        _In_ const sai_attribute_t *attr_list)
+{
+    DASH_LOG_ENTER();
+
+    *switch_id = DASH_MAKE_OID(SAI_OBJECT_TYPE_SWITCH,1),
+
+    DASH_LOG_NOTICE("created switch id: %ld", *switch_id);
 
-    *switch_id = MAKE_OID(SAI_OBJECT_TYPE_SWITCH,1),
-    fprintf(stderr, "sai_create_switch_dummy()\n");
     return SAI_STATUS_SUCCESS;
 }
 
-sai_object_id_t port_list[DASH_BMV2_NUM_PORTS] = {
-    MAKE_OID(SAI_OBJECT_TYPE_PORT,1),
-    MAKE_OID(SAI_OBJECT_TYPE_PORT,2)
-};
-
-sai_status_t sai_get_switch_attribute(
+static sai_status_t dash_sai_get_switch_attribute(
         _In_ sai_object_id_t switch_id,
         _In_ uint32_t attr_count,
-        _Inout_ sai_attribute_t *attr_list) {
+        _Inout_ sai_attribute_t *attr_list)
+{
+    DASH_LOG_ENTER();
+
+    DASH_LOG_NOTICE("get_switch_attribute()");
```

Comment:
Hhy not `dash_sai_get_switch_attribute()`?
Also, why both LOG calls, aren't they redundant?

Reply:
leftover from fprintf(stderr, "sai_get_switch_attribute()\n"); DASH_LOG_ENTER(); is logging always on debug, i can change the name